tests/test_qnode_legacy.py

```diff
@@ -1877,26 +1877,22 @@ def circuit():
 
         assert len(tapes) == 2
 
-    def test_invalid_hamiltonian_expansion_finite_shots(self):
-        """Test that an error is raised if multiple expectations are requested
-        when using finite shots"""
+    def test_multiple_hamiltonian_expansion_finite_shots(self):
+        """Test that multiple Hamiltonians works correctly (sum_expand should be used)"""
+
         dev = qml.device("default.qubit.legacy", wires=3, shots=50000)
 
         obs = [qml.PauliX(0), qml.PauliX(0) @ qml.PauliZ(1), qml.PauliZ(0) @ qml.PauliZ(1)]
         c = np.array([-0.6543, 0.24, 0.54])
         H = qml.Hamiltonian(c, obs)
-        H.compute_grouping()
-
-        assert len(H.grouping_indices) == 2
```

Comment:
would that break the ability to compute multiple hamiltonian expvals?

Reply:
I don't remember why I removed it, let me try adding it back.